### PR TITLE
Nuke: load color space from representation data

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Brief description
Supporting to load clip with colorspace set on representation data.

## Description
Colorspace data are firstly check on representation data and if there are none, then they are checked from version data. If there is found any it is validated if they are even supported by workfile set colorspace management by looking into available colorspace values. In case they are not, we look into settings in OP Image IO regex items. If this is not even set then the read node settings knob is not set at all and is reading workfile defaults set in root colorspace.

### CODE REVIEW

File: openpype/hosts/nuke/plugins/load/load_clip.py [hunk 1]
```diff
@@ -145,6 +142,13 @@ def load(self, context, name, namespace, options):
             for k in add_keys:
                 if k == 'version':
                     data_imprint.update({k: context["version"]['name']})
+                elif k == 'colorspace':
+                    colorspace = repre["data"].get(k)
+                    colorspace = colorspace or version_data.get(k)
+                    data_imprint.update({
+                        "db_colorspace": colorspace,
+                        "set_colorspace": set_colorspace
```
Comment (iLLiCiTiT) [inline openpype/hosts/nuke/plugins/load/load_clip.py:150]: What if `set_colorspace` or `colorspace` are set to `None`? Shouldn't it store empty string in that case?

Side note: From key `"set_colorspace"`is not obvious if it's verb or noun. I would change it to `"used_colorspace"` if it make sense.

Comment (jakubjezek001) [reply openpype/hosts/nuke/plugins/load/load_clip.py:150]: Good point, thanks ;)

**Review by iLLiCiTiT — APPROVED** [review]
Code looks ok. Unfotunatelly I don't know how to publish clip to test clip loader.

File: openpype/hosts/nuke/plugins/load/load_clip.py [hunk 2]
```diff
@@ -365,14 +372,38 @@ def _loader_shift(self, read_node, workfile_start=False):
...
+        colorspace = colorspace or version_data.get("colorspace")
+
+        # colorspace from `project_anatomy/imageio/nuke/regexInputs`
+        iio_colorspace = get_imageio_input_colorspace(path)
+
+        # Set colorspace defined in version data
+        if (
+            colorspace is not None
+            and colorspace_exists_on_node(
+                node, str(colorspace)) is not False
```
Comment (iLLiCiTiT) [inline openpype/hosts/nuke/plugins/load/load_clip.py:401]:
```suggestion
            and not colorspace_exists_on_node(node, str(colorspace))
```

**Pull request merged**